bin/propolis-server/src/lib/stats.rs

```diff
@@ -156,13 +154,27 @@ pub async fn start_oximeter_server(
                 return Ok(server);
             }
             Err(e) => {
-                if retry_print_timeout == 0 {
-                    error!(log, "Can't connect to oximeter server:\n{}", e);
-                    retry_print_timeout = 1;
+                if attempts >= N_RETRY {
+                    error!(
+                        log,
+                        "Could not connect to oximeter after {} retries",
+                        N_RETRY
+                    );
+                    return Err(e.into());
                 }
-                // Retry every 10 seconds, but only print once a minute
-                retry_print_timeout = (retry_print_timeout + 1) % 7;
-                tokio::time::sleep(tokio::time::Duration::from_secs(10)).await;
+                attempts += 1;
+
+                warn!(
```

Comment:
Should this `warn!()` go above line 157? If it's down here it won't capture the error code from the last call to `Server::start()`.

Reply:
should be irrelevant as of 2eb4619